coredb-operator/src/extensions.rs

```diff
@@ -378,12 +379,20 @@ fn extension_plan(have_changed: &[Extension], actual: &[Extension]) -> (Vec<Exte
         for extension_actual in actual {
             if extension_desired.name == extension_actual.name {
                 found = true;
-                if extension_desired != extension_actual {
-                    // if the extension exists, but the version is different, it needs to be dropped
-                    // and reinstalled
-                    changed.push(extension_desired.clone());
+                // extension exists, therefore has been installed
+                // determine if the `enabled` toggle has changed
+                'loc: for loc_desired in extension_desired.locations.clone() {
+                    for loc_actual in extension_actual.locations.clone() {
+                        if loc_desired.database == loc_actual.database {
+                            // TODO: when we want to support version changes, this is where we would do it
+                            if loc_desired.enabled != loc_actual.enabled {
+                                debug!("desired: {:?}, actual: {:?}", extension_desired, extension_actual);
+                                changed.push(extension_desired.clone());
+                                break 'loc;
+                            }
+                        }
+                    }
```

Comment:
this fixes a bug where we'd continuously reconcile an extension when the description of the extension has changed. We have no way to change the description of an extension, but we do change whether they are enabled. Therefore, only consider it a "change" if the enable toggle has changed, rather than any attribute about the extension.